src/base/NekRSSeparateDomainProblem.C

```diff
@@ -0,0 +1,343 @@
+#include "TimedPrint.h"
+#include "MooseUtils.h"
+#include "CardinalUtils.h"
+
+#include "nekrs.hpp"
+#include "nekInterface/nekInterfaceAdapter.hpp"
+
+registerMooseObject("CardinalApp", NekRSSeparateDomainProblem);
+
+bool NekRSSeparateDomainProblem::_first = true;
+
+InputParameters
+NekRSSeparateDomainProblem::validParams()
+{
+  InputParameters params = NekRSProblemBase::validParams();
+  params.addParam<bool>("minimize_transfers_in", false, "Whether to only synchronize nekRS "
+    "for the direction TO_EXTERNAL_APP on multiapp synchronization steps");
+  params.addParam<bool>("minimize_transfers_out", false, "Whether to only synchronize nekRS "
+    "for the direction FROM_EXTERNAL_APP on multiapp synchronization steps");
+  params.addParam<bool>("moving_mesh", false, "Moving mesh");
+  params.addParam<bool>("toNekRS_interface", false, "External app -> NekRS interface present?");
+  params.addParam<bool>("toNekRS_temperature", false, "External app -> NekRS temperature transfer?");
+  params.addParam<bool>("fromNekRS_interface", false, "NekRS -> external app interface present?");
+  params.addParam<bool>("fromNekRS_temperature", false, "NekRS -> external app temperature transfer?");
+  params.addRequiredParam<std::vector<FileName>>("ExternalApp_filename", "External app input file name");
+  MooseEnum app_type("SamApp THMApp");
+  params.addRequiredParam<MooseEnum>("ExternalApp_type", app_type, "External app type");
+  params.addParam<std::vector<int>>("fromNekRS_boundary", "Boundary ID through which nekRS will be coupled to external app");
+  params.addParam<std::vector<int>>("NekRS_inlet_boundary", "NekRS Boundary ID for Pressure drop calculation");
+
+  return params;
+}
+
+NekRSSeparateDomainProblem::NekRSSeparateDomainProblem(const InputParameters &params) : NekRSProblemBase(params),
+    _serialized_solution(NumericVector<Number>::build(_communicator).release()), 
+    _moving_mesh(getParam<bool>("moving_mesh")),
+    _minimize_transfers_in(getParam<bool>("minimize_transfers_in")),
+    _minimize_transfers_out(getParam<bool>("minimize_transfers_out")),
+    _toNekRS(getParam<bool>("toNekRS_interface")),
+    _toNekRS_temperature(getParam<bool>("toNekRS_temperature")),
+    _fromNekRS(getParam<bool>("fromNekRS_interface")),
+    _fromNekRS_temperature(getParam<bool>("fromNekRS_temperature")),
+    _ExternalApp_filename(getParam<std::vector<FileName>>("ExternalApp_filename")),
+    _ExternalApp_type(getParam<MooseEnum>("ExternalApp_type")),
+    _fromNekRS_boundary(isParamValid("fromNekRS_boundary") ? &getParam<std::vector<int>>("fromNekRS_boundary") : nullptr),
+    _NekRS_inlet_boundary(isParamValid("NekRS_inlet_boundary") ? &getParam<std::vector<int>>("NekRS_inlet_boundary") : nullptr)
+{
+
+  if (!_toNekRS && !_fromNekRS)
+    mooseError("This problem type needs atleast one of toNekRS_interface \n", 
+    "or fromNekRS_interface to be set to true\n");
+
+  if (_fromNekRS)
+    {
+    if (_fromNekRS_boundary->size() != 1)
```

Comment:
Hypothetical situation here - what if I have just one _physical_ inlet in my NekRS model, but for some reason it's split over two sidesets? As in, I have a pipe and the upper hemisphere entrance is boundary 1 and the lower hemisphere entrance is boundary 2. I think your algorithm should still work, correct? 

Is this error message here protecting against some fundamental issue, or just that we might need to adjust a function somewhere that takes a vector instead of a single value?

Reply:
`_fromNekRS_boundary` is only used to calculate a global pressure drop over the NekRS component, to use as a boundary condition in the system code (SAM, etc.). So the variable is only used in creating an average pressure postProcessor on the inlet boundary specified. The error ensures that there's only one inlet boundary used in the postprocessor. But your hypothetical situation would throw that error even if we didn't want it to.

Maybe we should actually keep this within the user's cardinal input file instead of automating the PostProcessor creation. It'll just require the user to generate a "pressure drop PostProcessor" by themselves for the nekRS component. That might be best, similar to removing all the automated MultiApp and transfers we have right now